mitiq/rem/rem.py

```diff
@@ -0,0 +1,182 @@
+# Copyright (C) 2021 Unitary Fund
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""Readout Confusion Inversion."""
+
+from typing import Callable, Optional, Union
+
+from functools import wraps
+import numpy as np
+
+from mitiq._typing import QPROGRAM, MeasurementResult
+from mitiq.executor.executor import Executor
+from mitiq.observable.observable import Observable
+
+
+def to_probability_vector(measurement: np.ndarray) -> np.ndarray:
+    """Convert a raw measurement to a probability vector.
+
+    Args:
+        measurement: A single measurement.
+    """
```

Comment:
I think we need a function that returns the probability vector associated to the full list of measurement results.
Something like:

```python
def bitstrings_to_probabiliy_vector(bitstrings: List[Bitstring]) -> np.ndarray:
    """Converts a list of measured bitstrings to a probability vector estimated as
the empirical frequency of each bitstring (ordered with increasing binary value). 
    Args:
        bitstrings: All measured bitstrings.
    """
```

The code of this new function is probably very similar to the code of your function but **averaged** over all bitstrings.

Reply:
This has been added.